core/templates/dev/head/exploration_editor/main_tab/StateEditor.js

```diff
@@ -77,35 +79,27 @@ oppia.controller('StateEditor', [
         $scope.isInteractionShown = true;
       }
 
+      $scope.$on('externalSave', function() {
+        if ($scope.contentEditorIsOpen) {
+          $scope.saveTextContent();
+        }
+      });
+
       $rootScope.loadingMessage = '';
     };
 
     $scope.openStateContentEditor = function() {
       if (editabilityService.isEditable()) {
-        $scope.contentMemento = angular.copy($scope.content);
+        $scope.contentEditorIsOpen = true;
       }
     };
 
     $scope.saveTextContent = function() {
```

Comment:
This seems to be at odds with ``openStateContentEditor``. Can we instead call it ``closeStateContentEditor``? It's natural for a close function to save its contents. It also has better parity with the open function and properly describes why ``contentEditorIsOpen`` is being set to ``false`` here.